Gems/Atom/Feature/Common/Code/Include/Atom/Feature/Mesh/StreamBufferViewsBuilderInterface.h

```diff
@@ -24,12 +25,36 @@ namespace AZ
-            virtual const AZ::RHI::Ptr<AZ::RHI::BufferView>& GetIndexBufferView() const = 0;
+            //! Returns the Shader-bindable RHI::BufferView for the Vertex Indices from a particular mesh.
+            //! REMARK: If the BindlessSrg is enabled, which typically is, this buffer view is already bound
+            //! as a ByteAddressBuffer
+            virtual const AZ::RHI::Ptr<AZ::RHI::BufferView>& GetVertexIndicesBufferView() const = 0;
+            //! Returns the StreamBufferView from the Vertex Indices, which is useful to know the actual
+            //! byte offset, byte count, etc related to the sub mesh(Mesh Index). 
+            virtual const AZ::RHI::IndexBufferView& GetVertexIndicesIndexBufferView() const = 0;
+            //! A shortcut to get the Read Index from the BindlessSrg for the IndexBufferView.
+            //! Alternatively, you could get the same index by getting it directly from the AZ::RHI::BufferView
+            //! returned by GetIndexBufferView()
+            virtual uint32_t GetVertexIndicesBindlessReadIndex(int deviceIndex) const = 0;
+
             //! Returns the Shader-bindable RHI::BufferView for a Vertex Stream, identifiable by @shaderSemantic, from a particular mesh.
-            virtual const AZ::RHI::Ptr<AZ::RHI::BufferView>& GetStreamBufferView(const AZ::RHI::ShaderSemantic& shaderSemantic) const = 0;
+            virtual const AZ::RHI::Ptr<AZ::RHI::BufferView>& GetBufferView(const AZ::RHI::ShaderSemantic& shaderSemantic) const = 0;
             //! Same as above, but provides the convenience of finding the vertex stream by name, like "POSITION" or "UV1", etc.
-            virtual const AZ::RHI::Ptr<AZ::RHI::BufferView>& GetStreamBufferView(const char* semanticName) const = 0;
+            virtual const AZ::RHI::Ptr<AZ::RHI::BufferView>& GetBufferView(const char* semanticName) const = 0;
+
+            //! Returns the StreamBufferView for a Vertex Stream, identifiable by @shaderSemantic, from a particular mesh.
+            //! This is useful to know the actual byte offset, byte count, etc related to the sub mesh(Mesh Index).
+            //! if the semantic is not fund, it returns nullptr.
+            virtual const AZ::RHI::StreamBufferView* GetStreamBufferView(const AZ::RHI::ShaderSemantic& shaderSemantic) const = 0;
+            //! Same as above, but provides the convenience of finding the vertex stream by name, like "POSITION" or "UV1", etc.
+            virtual const AZ::RHI::StreamBufferView* GetStreamBufferView(const char* semanticName) const = 0;
+
+            //! A shortcut to get the Read Index from the BindlessSrg for a particular BufferView identifiable by its shader semantic.
+            //! Alternatively, you could get the same index by getting it directly from the AZ::RHI::BufferView
+            //! returned by GetBufferView().
+            virtual uint32_t GetStreamBufferViewBindlessReadIndex(int deviceIndex, const AZ::RHI::ShaderSemantic& shaderSemantic) const = 0;
```

Comment:
Same as above, can you add a function without the device-index, that returns a map with the read-index per device?

Reply:
Done.